### ISSUE TITLE
Deprecate old ways of creating ParameterValues

### ISSUE THREAD

**Issue by tinosulzer** [pybamm-team/PyBaMM#2901]
I think the transition to using python files or json file (BPX) for parameter values has gone fairly well so we can start to deprecate some of the old methods for importing values from csv files. These are no longer recommended but still possible from the codebase. We should also update the training videos on this very soon.
- [ ] remove functionality to update from chemistry (old approach with separate files for anode/cathode/electrolyte etc)
- [ ] remove functionality to load from csv
- [ ] remove functionality to load a function from a string "[function]function_name"
- [ ] remove functionality to load data from a string "[data]data_name"

Basically, users will be responsible for passing in a python dictionary where functions are python functions. Interpolants can be created as functions [like this](https://github.com/pybamm-team/PyBaMM/blob/18d4e84d07a8afb1d8a71081c2912f486b11283d/pybamm/input/parameters/lithium_ion/Ai2020.py#L501-L517)

This is a breaking change so let's discuss before committing

**Comment by brosaplanella** [comment]
Shall we keep these options with a warning for a few releases and then remove?

**Comment by tinosulzer** [comment]
We've had this warning up for ~6 releases https://github.com/pybamm-team/PyBaMM/blob/18d4e84d07a8afb1d8a71081c2912f486b11283d/pybamm/parameters/parameter_values.py#L65

**Comment by brosaplanella** [comment]
Then happy to deprecate it :)
